sklearn/metrics/cluster/_supervised.py

```diff
@@ -1312,3 +1307,12 @@ def entropy(labels):
     # Always convert the result as a Python scalar (on CPU) instead of a device
     # specific scalar array.
     return float(-xp.sum((pi / pi_sum) * (xp.log(pi) - log(pi_sum))))
+
+
+# TODO(1.10): Remove
+@deprecated(
+    "`entropy` is deprecated in 1.8 and will become a private function only from 1.10."
+)
+def entropy(labels):
+    """Public version of `_entropy`. To be deprecated."""
```

Comment:
I didn't add a full docstring because I don't think we ever created an API entry for it.